cristin-project/src/main/java/no/unit/nva/cristin/projects/query/QueryCristinProjectApiClient.java

```diff
@@ -1,96 +1,60 @@
-import static no.unit.nva.cristin.model.JsonPropertyNames.QUERY;
-import static no.unit.nva.utils.UriUtils.PROJECT;
-import static no.unit.nva.utils.UriUtils.createIdUriFromParams;
-import static no.unit.nva.utils.UriUtils.queryParameters;
-import static nva.commons.core.attempt.Try.attempt;
 import java.net.URI;
-import java.net.URLEncoder;
 import java.net.http.HttpResponse;
-import java.nio.charset.StandardCharsets;
-import java.util.List;
-import java.util.Map;
-import no.unit.nva.cristin.common.Utils;
-import no.unit.nva.cristin.model.Constants.QueryType;
+import no.unit.nva.cristin.model.Constants.QueryParameterKey;
 import no.unit.nva.cristin.model.SearchResponse;
 import no.unit.nva.cristin.projects.common.CristinProjectApiClient;
 import no.unit.nva.cristin.projects.common.CristinQuery;
-import no.unit.nva.cristin.projects.model.cristin.CristinProject;
 import no.unit.nva.cristin.projects.model.nva.NvaProject;
 import nva.commons.apigateway.exceptions.ApiGatewayException;
 
 public class QueryCristinProjectApiClient extends CristinProjectApiClient {
 
+
     /**
      * Creates a wrapper object containing Cristin Projects transformed to NvaProjects with additional metadata. Is used
      * for serialization to the client.
      *
-     * @param requestQueryParameters Request parameters from client containing title and language
+     * @param cristinQuery CristinQuery from client containing title and language
      * @return a SearchResponse filled with transformed Cristin Projects and metadata
      * @throws ApiGatewayException if some errors happen we should return this to client
      */
     public SearchResponse<NvaProject> queryCristinProjectsIntoWrapperObjectWithAdditionalMetadata(
-        Map<String, String> requestQueryParameters) throws ApiGatewayException {
+        CristinQuery cristinQuery) throws ApiGatewayException {
+
+        final var startRequestTime = System.currentTimeMillis();
+
+        final var response = queryProjects(cristinQuery);
+        var cristinProjects =
+            getEnrichedProjectsUsingQueryResponse(response, cristinQuery.getValue(QueryParameterKey.LANGUAGE));
 
-        long startRequestTime = System.currentTimeMillis();
-        QueryType queryType = getQueryTypeBasedOnParameters(requestQueryParameters);
-        HttpResponse<String> response = queryProjects(requestQueryParameters, queryType);
-        List<CristinProject> cristinProjects =
-            getEnrichedProjectsUsingQueryResponse(response, requestQueryParameters.get(LANGUAGE));
-        if (cristinProjects.isEmpty() && queryType == QUERY_USING_GRANT_ID) {
-            response = queryProjects(requestQueryParameters, QUERY_USING_TITLE);
-            cristinProjects = getEnrichedProjectsUsingQueryResponse(response, requestQueryParameters.get(LANGUAGE));
+        if (cristinProjects.isEmpty()) { // && queryParameterKey == GRANT_ID) {
+            cristinProjects = getEnrichedProjectsUsingQueryResponse(
```

Comment:
This is duplicate. The reason it was called twice before is because if grantid was the query type and no items were found, we did a new search using title. Now i understand they are their own parameters therefore we only need to query once